curriculum/challenges/english/blocks/lecture-working-with-npm-scripts/692aba0978ca391f10455b4a.md

```diff
@@ -0,0 +1,327 @@
+
+```bash
+https://www.npmjs.com/package/<your-package-name>
+```
+
+If there isn't a package at that URL, the package name is available and you can use it.
+
+However, if the name is already taken and you really want to keep it, you can use what is known as a **scope** to publish it.
+
+### **Scopes**
+
+When you create a user or organization account in the npm registry, you are granted a scope with your username or the name of your organization.
+
+Scopes are helpful to avoid name conflicts because you can publish your package under your scope, even if another package with the same name already exists.
+
+The name of a scoped package starts with the scope name. For example, in `package.json`, you would find this key-value pair:
+
+```bash
+"name": "@<scope-name>/<your-package-name>"
+```
+
+You can make this change directly in the file if you need to, or you can create the `package.json` file with the scope from the start by adding `--scope=@<scope-name>` to the `npm init` command.
+
+```bash
+npm init --scope=@<scope-name>
+```
+
+The scope name can be your username if you are using a personal account, or the name of your organization if you are using an organizational account.
+
+## Step 4: Create README.md
+
+The README of your package is a markdown file where you describe what your package does and its intended usage.
+
+Some examples of important content that you may want to include are:
+
+* Its purpose.
+    
+* How to install it.
+    
+* Important dependencies and considerations.
+    
+* How to get started.
+    
+* Sample code snippets.
+    
+* Contribution guidelines (if you accept contributions).
+    
+* Link to a changelog file where you track the changes that have been made in different versions.
+    
+
+The structure of the README is flexible, though. You can adapt it to fit your needs.
+
+## **Step 5:** `.npmignore`
+
+The `.npmignore` file tells the npm registry what files to include and what files to ignore when publishing your package. It's very important to prevent sharing sensitive information such as private keys, passwords, and personal information.
+
+In this file, you can write patterns to specify what files, directories, and types of files are ignored. It follows the same pattern matching rules as the `.gitignore` files. You can learn more about these rules in the [official Git documentation](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository#_ignoring).
+
+If you have worked with Git, you may think that this file sounds very similar to `.gitignore`, right? Well, that's because it is. According to the [npm documentation](https://docs.npmjs.com/cli/v9/using-npm/developers), "If there's no `.npmignore` file, but there is a `.gitignore` file, then npm will ignore the stuff matched by the `.gitignore` file."
```

Comment:
```suggestion
In this file, you can write patterns to specify what files, directories, and types of files are ignored. It follows the same pattern matching rules as the `.gitignore` files.

If you have worked with Git, you may think that this file sounds very similar to `.gitignore`, right? Well, that's because it is. According to the npm documentation, "If there's no `.npmignore` file, but there is a `.gitignore` file, then npm will ignore the stuff matched by the `.gitignore` file."
```

Generally, we don't like links to external sites. I would vote to just remove them, but I would be fine leaving them if we think they're necessary. I think it works without them.